# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1544745697 28800
#      Thu Dec 13 16:01:37 2018 -0800
# Node ID 3af4f3e2dae865bc159a49b577111b9cb53143e3
# Parent  4f044bc43f07e1c5e2445d93e52db683ff2919bb
remove unsafe "SED" from the build env stored in liboctinterp and mkoctfile

* build-env.h, build-env.in.cc (octave::build_env::SED): Remove variable.
* toplev.cc (F__octave_config_info__): Remove "SED" from build_environment.
* mkoctfile.in.cc: Remove "SED" from vars and help_msg.
* mkoctfile.1, mkoctfile.m: Remove "SED" from documentation.
* subst-config-vals.in.sh, subst-cross-config-vals.in.sh (OCTAVE_CONF_SED):
Remove from the list of substitution variables populated from the build env.

diff --git a/build-aux/subst-config-vals.in.sh b/build-aux/subst-config-vals.in.sh
--- a/build-aux/subst-config-vals.in.sh
+++ b/build-aux/subst-config-vals.in.sh
@@ -199,17 +199,16 @@ QRUPDATE_LDFLAGS="@QRUPDATE_LDFLAGS@"
 QRUPDATE_LIBS="@QRUPDATE_LIBS@"
 QT_CPPFLAGS="@QT_CPPFLAGS@"
 QT_LDFLAGS="@QT_LDFLAGS@"
 QT_LIBS="@QT_LIBS@"
 QT_OPENGL_LIBS="@QT_OPENGL_LIBS@"
 RANLIB="@RANLIB@"
 RDYNAMIC_FLAG="@RDYNAMIC_FLAG@"
 READLINE_LIBS="@READLINE_LIBS@"
-SED="@SED@"
 SHARED_LIBS="@SHARED_LIBS@"
 SH_LD="@SH_LD@"
 SH_LDFLAGS="@SH_LDFLAGS@"
 STATIC_LIBS="@STATIC_LIBS@"
 SUITESPARSECONFIG_LIBS="@SUITESPARSECONFIG_LIBS@"
 SUNDIALS_IDA_CPPFLAGS="@SUNDIALS_IDA_CPPFLAGS@"
 SUNDIALS_IDA_LDFLAGS="@SUNDIALS_IDA_LDFLAGS@"
 SUNDIALS_IDA_LIBS="@SUNDIALS_IDA_LIBS@"
@@ -373,17 +372,16 @@ config_opts="@config_opts@"
   -e "s|%OCTAVE_CONF_QT_CPPFLAGS%|\"${QT_CPPFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_QT_LDFLAGS%|\"${QT_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_QT_LIBS%|\"${QT_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_QT_OPENGL_LIBS%|\"${QT_OPENGL_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_RANLIB%|\"${RANLIB}\"|" \
   -e "s|%OCTAVE_CONF_RDYNAMIC_FLAG%|\"${RDYNAMIC_FLAG}\"|" \
   -e "s|%OCTAVE_CONF_READLINE_LIBS%|\"${READLINE_LIBS}\"|" | \
   $SED \
-  -e "s|%OCTAVE_CONF_SED%|\"${SED}\"|" \
   -e "s|%OCTAVE_CONF_SHARED_LIBS%|\"${SHARED_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_SH_LD%|\"${SH_LD}\"|" \
   -e "s|%OCTAVE_CONF_SH_LDFLAGS%|\"${SH_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_STATIC_LIBS%|\"${STATIC_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_SUITESPARSECONFIG_LIBS%|\"${SUITESPARSECONFIG_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_SUNDIALS_IDA_CPPFLAGS%|\"${SUNDIALS_IDA_CPPFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_SUNDIALS_IDA_LDFLAGS%|\"${SUNDIALS_IDA_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_SUNDIALS_IDA_LIBS%|\"${SUNDIALS_IDA_LIBS}\"|" \
diff --git a/build-aux/subst-cross-config-vals.in.sh b/build-aux/subst-cross-config-vals.in.sh
--- a/build-aux/subst-cross-config-vals.in.sh
+++ b/build-aux/subst-cross-config-vals.in.sh
@@ -203,17 +203,16 @@ QRUPDATE_LDFLAGS="@QRUPDATE_LDFLAGS@"
 QRUPDATE_LIBS="@QRUPDATE_LIBS@"
 QT_CPPFLAGS="@QT_CPPFLAGS@"
 QT_LDFLAGS="@QT_LDFLAGS@"
 QT_LIBS="@QT_LIBS@"
 QT_OPENGL_LIBS="@QT_OPENGL_LIBS@"
 RANLIB="@RANLIB@"
 RDYNAMIC_FLAG="@RDYNAMIC_FLAG@"
 READLINE_LIBS="@READLINE_LIBS@"
-SED="@SED@"
 SHARED_LIBS="@SHARED_LIBS@"
 SH_LD="@SH_LD@"
 SH_LDFLAGS="@SH_LDFLAGS@"
 STATIC_LIBS="@STATIC_LIBS@"
 SUITESPARSECONFIG_LIBS="@SUITESPARSECONFIG_LIBS@"
 SUNDIALS_IDA_CPPFLAGS="@SUNDIALS_IDA_CPPFLAGS@"
 SUNDIALS_IDA_LDFLAGS="@SUNDIALS_IDA_LDFLAGS@"
 SUNDIALS_IDA_LIBS="@SUNDIALS_IDA_LIBS@"
@@ -377,17 +376,16 @@ config_opts="@config_opts@"
   -e "s|%OCTAVE_CONF_QT_CPPFLAGS%|\"${QT_CPPFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_QT_LDFLAGS%|\"${QT_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_QT_LIBS%|\"${QT_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_QT_OPENGL_LIBS%|\"${QT_OPENGL_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_RANLIB%|\"${RANLIB}\"|" \
   -e "s|%OCTAVE_CONF_RDYNAMIC_FLAG%|\"${RDYNAMIC_FLAG}\"|" \
   -e "s|%OCTAVE_CONF_READLINE_LIBS%|\"${READLINE_LIBS}\"|" | \
   $SED \
-  -e "s|%OCTAVE_CONF_SED%|\"${SED}\"|" \
   -e "s|%OCTAVE_CONF_SHARED_LIBS%|\"${SHARED_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_SH_LD%|\"${SH_LD}\"|" \
   -e "s|%OCTAVE_CONF_SH_LDFLAGS%|\"${SH_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_STATIC_LIBS%|\"${STATIC_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_SUITESPARSECONFIG_LIBS%|\"${SUITESPARSECONFIG_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_SUNDIALS_IDA_CPPFLAGS%|\"${SUNDIALS_IDA_CPPFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_SUNDIALS_IDA_LDFLAGS%|\"${SUNDIALS_IDA_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_SUNDIALS_IDA_LIBS%|\"${SUNDIALS_IDA_LIBS}\"|" \
diff --git a/doc/interpreter/mkoctfile.1 b/doc/interpreter/mkoctfile.1
--- a/doc/interpreter/mkoctfile.1
+++ b/doc/interpreter/mkoctfile.1
@@ -109,19 +109,19 @@ Print configuration variable \fIVAR\fP. 
     CXXFLAGS                  LIBOCTINTERP
     CXXPICFLAG                LIBS
     DEPEND_EXTRA_SED_PATTERN  OCTAVE_LIBS
     DEPEND_FLAGS              OCTAVE_LINK_DEPS
     DL_LD                     OCT_LINK_DEPS
     DL_LDFLAGS                RANLIB
     EXEEXT                    RDYNAMIC_FLAG
     F77                       READLINE_LIBS
-    F77_INTEGER_8_FLAG        SED
-    FFLAGS                    XTRA_CFLAGS
-    FFTW3_LDFLAGS             XTRA_CXXFLAGS
+    F77_INTEGER_8_FLAG        XTRA_CFLAGS
+    FFLAGS                    XTRA_CXXFLAGS
+    FFTW3_LDFLAGS
     FFTW3_LIBS
 .Ve
 .RE
 .TP
 .B \-v\fR,\fB --verbose
 Echo commands as they are executed.
 .TP
 .B file
diff --git a/libinterp/build-env.h b/libinterp/build-env.h
--- a/libinterp/build-env.h
+++ b/libinterp/build-env.h
@@ -149,17 +149,16 @@ namespace octave
     extern const char *QRUPDATE_LIBS;
     extern const char *QT_CPPFLAGS;
     extern const char *QT_LDFLAGS;
     extern const char *QT_LIBS;
     extern const char *QT_OPENGL_LIBS;
     extern const char *RANLIB;
     extern const char *RDYNAMIC_FLAG;
     extern const char *READLINE_LIBS;
-    extern const char *SED;
     extern const char *SHARED_LIBS;
     extern const char *SH_LD;
     extern const char *SH_LDFLAGS;
     extern const char *STATIC_LIBS;
     extern const char *SUITESPARSECONFIG_LIBS;
     extern const char *SUNDIALS_IDA_CPPFLAGS;
     extern const char *SUNDIALS_IDA_LDFLAGS;
     extern const char *SUNDIALS_IDA_LIBS;
diff --git a/libinterp/build-env.in.cc b/libinterp/build-env.in.cc
--- a/libinterp/build-env.in.cc
+++ b/libinterp/build-env.in.cc
@@ -268,18 +268,16 @@ namespace octave
     const char *QT_OPENGL_LIBS = %OCTAVE_CONF_QT_OPENGL_LIBS%;
 
     const char *RANLIB = %OCTAVE_CONF_RANLIB%;
 
     const char *RDYNAMIC_FLAG = %OCTAVE_CONF_RDYNAMIC_FLAG%;
 
     const char *READLINE_LIBS = %OCTAVE_CONF_READLINE_LIBS%;
 
-    const char *SED = %OCTAVE_CONF_SED%;
-
     const char *SHARED_LIBS = %OCTAVE_CONF_SHARED_LIBS%;
 
     const char *SH_LD = %OCTAVE_CONF_SH_LD%;
 
     const char *SH_LDFLAGS = %OCTAVE_CONF_SH_LDFLAGS%;
 
     const char *STATIC_LIBS = %OCTAVE_CONF_STATIC_LIBS%;
 
diff --git a/libinterp/corefcn/toplev.cc b/libinterp/corefcn/toplev.cc
--- a/libinterp/corefcn/toplev.cc
+++ b/libinterp/corefcn/toplev.cc
@@ -546,17 +546,16 @@ specified option.
            { "QRUPDATE_LIBS", octave::build_env::QRUPDATE_LIBS },
            { "QT_CPPFLAGS", octave::build_env::QT_CPPFLAGS },
            { "QT_LDFLAGS", octave::build_env::QT_LDFLAGS },
            { "QT_LIBS", octave::build_env::QT_LIBS },
            { "QT_OPENGL_LIBS", octave::build_env::QT_OPENGL_LIBS },
            { "RANLIB", octave::build_env::RANLIB },
            { "RDYNAMIC_FLAG", octave::build_env::RDYNAMIC_FLAG },
            { "READLINE_LIBS", octave::build_env::READLINE_LIBS },
-           { "SED", octave::build_env::SED },
            { "SHARED_LIBS", octave::build_env::SHARED_LIBS },
            { "SH_LD", octave::build_env::SH_LD },
            { "SH_LDFLAGS", octave::build_env::SH_LDFLAGS },
            { "STATIC_LIBS", octave::build_env::STATIC_LIBS },
            { "SUITESPARSECONFIG_LIBS", octave::build_env::SUITESPARSECONFIG_LIBS },
            { "TERM_LIBS", octave::build_env::TERM_LIBS },
            { "UMFPACK_CPPFLAGS", octave::build_env::UMFPACK_CPPFLAGS },
            { "UMFPACK_LDFLAGS", octave::build_env::UMFPACK_LDFLAGS },
diff --git a/scripts/miscellaneous/mkoctfile.m b/scripts/miscellaneous/mkoctfile.m
--- a/scripts/miscellaneous/mkoctfile.m
+++ b/scripts/miscellaneous/mkoctfile.m
@@ -95,20 +95,20 @@
 ##    DL_LD                       OCTAVE_PREFIX
 ##    DL_LDFLAGS                  OCTINCLUDEDIR
 ##    F77                         OCTLIBDIR
 ##    F77_INTEGER8_FLAG           OCT_LINK_DEPS
 ##    FFLAGS                      OCT_LINK_OPTS
 ##    FFTW3F_LDFLAGS              RANLIB
 ##    FFTW3F_LIBS                 RDYNAMIC_FLAG
 ##    FFTW3_LDFLAGS               READLINE_LIBS
-##    FFTW3_LIBS                  SED
-##    FFTW_LIBS                   SPECIAL_MATH_LIB
-##    FLIBS                       XTRA_CFLAGS
-##    FPICFLAG                    XTRA_CXXFLAGS
+##    FFTW3_LIBS                  SPECIAL_MATH_LIB
+##    FFTW_LIBS                   XTRA_CFLAGS
+##    FLIBS                       XTRA_CXXFLAGS
+##    FPICFLAG
 ## @end example
 ##
 ## @item --link-stand-alone
 ## Link a stand-alone executable file.
 ##
 ## @item --mex
 ## Assume we are creating a MEX file.  Set the default output extension to
 ## ".mex".
diff --git a/src/mkoctfile.in.cc b/src/mkoctfile.in.cc
--- a/src/mkoctfile.in.cc
+++ b/src/mkoctfile.in.cc
@@ -180,18 +180,16 @@ initialize (void)
   vars["MAN1DIR"] = prepend_octave_home (%OCTAVE_MAN1DIR%);
   vars["MANDIR"] = prepend_octave_home (%OCTAVE_MANDIR%);
   vars["OCTDATADIR"] = prepend_octave_home (%OCTAVE_OCTDATADIR%);
   vars["OCTDOCDIR"] = prepend_octave_home (%OCTAVE_OCTDOCDIR%);
   vars["OCTFILEDIR"] = prepend_octave_exec_home (%OCTAVE_OCTFILEDIR%);
   vars["OCTFONTSDIR"] = prepend_octave_home (%OCTAVE_OCTFONTSDIR%);
   vars["STARTUPFILEDIR"] = prepend_octave_home (%OCTAVE_STARTUPFILEDIR%);
 
-  vars["SED"] = get_variable ("SED", %OCTAVE_CONF_SED%);
-
   vars["OCTINCLUDEDIR"]
     = get_variable ("OCTINCLUDEDIR",
                     prepend_octave_home (%OCTAVE_CONF_OCTINCLUDEDIR%));
 
   vars["INCLUDEDIR"]
     = get_variable ("INCLUDEDIR",
                     prepend_octave_home (%OCTAVE_CONF_INCLUDEDIR%));
 
@@ -420,17 +418,16 @@ static std::string help_msg =
 "                             FFTW3F_LIBS\n"
 "                             FFTW3_LDFLAGS\n"
 "                             FFTW3_LIBS\n"
 "                             FFTW_LIBS\n"
 "                             FLIBS\n"
 "                             LIBS\n"
 "                             RANLIB\n"
 "                             READLINE_LIBS\n"
-"                             SED\n"
 "\n"
 "                          Octave configuration variables that are provided\n"
 "                          for informational purposes only.  Except for\n"
 "                          OCTAVE_HOME and OCTAVE_EXEC_HOME, users may not\n"
 "                          override these variables.\n"
 "\n"
 "                          If OCTAVE_HOME or OCTAVE_EXEC_HOME are set in\n"
 "                          the environment, then other variables are adjusted\n"
